stl/inc/xstring

```diff
@@ -2642,6 +2642,47 @@ private:
     }
 
 public:
+#if _HAS_CXX20
+    void _Move_assign_from_buffer(_Elem* const _Right, const size_type _Size, const size_type _Res) {
+        // Move assign from a buffer, used by basic_stringbuf
+        _Tidy_deallocate();
+        _Traits::assign(_Right[_Size], _Elem());
+        pointer _Fancy_right = pointer_traits<pointer>::pointer_to(*_Right);
+        auto& _My_data       = _Mypair._Myval2;
+        _My_data._Mysize     = _Size;
+        _My_data._Myres      = _Res;
+        if (_My_data._Large_string_engaged()) {
+            _Construct_in_place(_My_data._Bx._Ptr, _Fancy_right);
+        } else {
+            _Traits::copy(_My_data._Bx._Buf, _Right, _Res);
+            _Getal().deallocate(_Fancy_right, _Res);

```

Comment:
This is technically correct but it would be nice to leave ownership of the buffer with the stream in this case.

Reply:
I cannot find a way to leave ownership with the stream, as it doesn't know whether the buffer is large enough to avoid SSO. Also at this time, the allocator is moved to the string, so it is OK for the string to manage the buffer.

Reply:
>I cannot find a way to leave ownership with the stream

By having the stream check whether the resulting stream would be _Large_string_engaged rather than deferring to the string constructor, or creating a protocol whereby the string constructor could signal whether it took ownership via an out parameter.